roles/v2/mining-proxy/src/lib/downstream_mining.rs

```diff
@@ -306,6 +313,23 @@ impl DownstreamMiningNode {
     }
 
     pub fn exit(self_: Arc<Mutex<Self>>) {
+        if let Some(up) = self_.safe_lock(|s| s.upstream.clone()).unwrap() {
+            super::upstream_mining::UpstreamMiningNode::remove_dowstream(up,self_.clone());
+            //if let Some(id) = self_.safe_lock(|d| {
```

Comment:
Were these comments intentionally left here? If so, perhaps some context for what they might be for could be useful.